spanish-translations.md

```diff
@@ -0,0 +1,85 @@
+We got to use them all!
+
+By the way, this is how the API call to translate a script file looks like:
+```php
+public function translate(string $text, string $glossaryId = null): string
+{
+    $response = $this->deepL->translate(
+        $text,
+        Locale::LOCALE_DEFAULT, // source language
+        Locale::LOCALE_SPANISH, // target language
+        'xml', // tag handling
+        static::getIgnoredTags(),
+        'less', // formality
+        'nonewlines', // split sentences
+        1, // preserver formatting
+        static::getNonSplittingTags(),
+        static::getSplittingTags(),
+        $glossaryId
+    );
+
+    return $response[0]['text'];
+}
+```
+
+After translating the content we run a sort of inverse process for restoring the 
+XML tags into Markdown, besides fixing a few flaws that comes from DeepL.
+
+## DeepL Glossaries
+
+The API also comes with a glossary where you can define a set of words that you want to translate
+in a particular way. In our case, is very useful for not translating library names. 
+
+Fun fact: _Twig_ translates into _Ramita_!
+
+## Script-clicking Feature
+
+You may have noticed that if you click on the script text the video will start playing
+at that specific moment. We manage to do so by synchronizing word by word the script 
+with the subtitles, so it's very important to keep both files as similar as possible. This
+turned out to be a problem because we generate the Spanish subtitles from the output
+of an AI algorithm (which is not always idempotent). So, what we do is to run this script-clicking
+on the translated subtitles, and in case of an error, we send an alert so someone from the staff
+can fix it, unfortunately this latter step is still a manual process.
+
+## The `StofDoctrineExtensionsBundle` comes in to play
+
+One thing is to accurately translate a tutorial but delivering a multi-language site
+it's quite a different thing to do. As you may have noticed we partially translated the site
+into Spanish, you will see Spanish content only on those tutorials that are translated. For this
+purpose we installed and configured the `StofDoctrineExtensionsBundle` bundle in the application. 
+If you can learn more about the _translatable_ behavior [here](https://github.com/doctrine-extensions/DoctrineExtensions/blob/main/doc/translatable.md)
+
+## Want to Help?
+
+Since this is a process run by machines, you'll notice some "misprints" or just... silly translations. 
+Machines haven't quite taken over the world yet. But that's ok because we've also delivered a way 
+where you can easily help improve or fix an error by sending a pull request to a public GitHub repository.
+You only need to click on the "Edit on GitHub" button which is at the right top corner of the script area, 
+edit the script (or caption) file and commit the changes.
+
+Oh, and right now, we're focused only on translating content into Spanish, but we may expand 
+in the future to other languages.
```

Comment:
```suggestion
to other languages in the future.
```

Reply:
Honestly I don't know what option is better. I'll let Ryan to decide it :)

Reply:
I believe it's "more correct" grammatically, it's the same as "today/yesterday", i.e. means a point in time - you usually put it at the very end of the sentence, or in the very beginning :)

Reply:
Fair enough!